client/src/components/Header/components/nav-links.tsx

```diff
@@ -50,6 +53,47 @@ const mapDispatchToProps = {
   openSignoutModal
 };
 
+interface DonateButtonProps {
+  isUserDonating: boolean | undefined;
+  handleMenuKeyDown: (event: React.KeyboardEvent<HTMLAnchorElement>) => void;
+  t: TFunction;
+}
+
+const DonateButton = ({
+  isUserDonating,
+  handleMenuKeyDown,
+  t
+}: DonateButtonProps) => {
+  const exposeUniversalDonateButton = useFeature('expose_donate_button').on;
+  const donateItem = (
+    <li key='donate'>
+      <Link
+        className='nav-link'
+        onKeyDown={handleMenuKeyDown}
+        sameTab={false}
+        to='/donate'
+        data-test-label='dropdown-donate-button'
+        nav-donate-button
+      >
+        {t('buttons.donate')}
+      </Link>
+    </li>
+  );
+  const thankYouItem = (
+    <li key='donate'>
+      <div className='nav-link nav-link-flex nav-link-header'>
+        <span>{t('donate.thanks')}</span>
+        <FontAwesomeIcon icon={faHeart} />
+      </div>
+    </li>
+  );
+
+  if (isUserDonating) return thankYouItem;
+  else if (exposeUniversalDonateButton)
+    return <Media maxWidth={DONATE_NAV_EXPOSED_WIDTH}>{donateItem}</Media>;
+  else return donateItem;
+};
```

Comment:
this was hard to read, and it didn't look like React code, I don't know if it's actually re-rendering the `thankyouItem` and `donateItem`, because I can't change the state consistently.

For now this is a suggestion for readability and I will test the could in sandbox or something for the re-render info

```suggestion
type DonateItemProps = Pick<DonateButtonProps, 'handleMenuKeyDown'> & {
  donateText: string
}

const DonateItem = ({handleMenuKeyDown, donateText}: DonateItemProps) => (
  <li key='donate'>
    <Link
      className='nav-link'
      onKeyDown={handleMenuKeyDown}
      sameTab={false}
      to='/donate'
      data-test-label='dropdown-donate-button'
      nav-donate-button
    >
      {donateText}
    </Link>
  </li>
);

const ThankYouMessage = ({message}: {message: string}) => (
  <li key='donate'>
    <div className='nav-link nav-link-flex nav-link-header'>
      <span>{message}</span>
      <FontAwesomeIcon icon={faHeart} />
    </div>
  </li>
);

const DonateButton = ({
  isUserDonating,
  handleMenuKeyDown,
  t
}: DonateButtonProps) => {
  const exposeUniversalDonateButton = useFeature('expose_donate_button').on; 
  if (isUserDonating) return <ThankYouMessage message={t('donate.thanks')} />;
  else if (exposeUniversalDonateButton)
    return <Media maxWidth={DONATE_NAV_EXPOSED_WIDTH}><DonateItem handleMenuKeyDown={handleMenuKeyDown} donateText={t('buttons.donate')} /></Media>;
  else return <DonateItem handleMenuKeyDown={handleMenuKeyDown} donateText={t('buttons.donate')} />;
};
```